dipy/denoise/tests/test_kernel.py

```diff
@@ -0,0 +1,78 @@
+from dipy.denoise.enhancement_kernel import EnhancementKernel
+from dipy.denoise.shift_twist_convolution import convolve_sf
+from dipy.reconst.shm import sh_to_sf, sf_to_sh
+from dipy.data import get_sphere
+
+import numpy as np
+import numpy.testing as npt
+
+def test_enhancement_kernel():
+    """ Test if the kernel values are correct, at the same time check if the 
+    the lookup-table of the kernel."""
+
+    D33 = 1.0
+    D44 = 0.04
+    t = 1
+    k = EnhancementKernel(D33, D44, t, test_mode=True)
+
+    y = np.array([0., 0., 0.])
+    v = np.array([0., 0., 1.])
+    orientationlist=[[0., 0., 1.], [-0.0527864, 0.688191, 0.723607], \
+      [-0.67082, -0.16246, 0.723607], [-0.0527864, -0.688191, 0.723607], \
+      [0.638197, -0.262866, 0.723607], [0.831052, 0.238856, 0.502295], \
+      [0.262866, -0.809017, -0.525731], [0.812731, 0.295242, -0.502295], \
+      [-0.029644, 0.864188, -0.502295], [-0.831052, 0.238856, -0.502295], \
+      [-0.638197, -0.262866, -0.723607], [-0.436009, 0.864188, -0.251148], \
+      [-0.687157, -0.681718, 0.251148], [0.67082, -0.688191, 0.276393], \
+      [0.67082, 0.688191, 0.276393], [0.947214, 0.16246, -0.276393], \
+      [-0.861803, -0.425325, -0.276393]]
+    positionlist= [[-0.108096, 0.0412229, 0.339119], [0.220647, -0.422053, 0.427524], \
+      [-0.337432, -0.0644619, -0.340777], [0.172579, -0.217602, -0.292446], \
+      [-0.271575, -0.125249, -0.350906], [-0.483807, 0.326651, 0.191993], \
+      [-0.480936, -0.0718426, 0.33202], [0.497193, -0.00585659, -0.251344], \
+      [0.237737, 0.013634, -0.471988], [0.367569, -0.163581, 0.0723955], \
+      [0.47859, -0.143252, 0.318579], [-0.21474, -0.264929, -0.46786], \
+      [-0.0684234, 0.0342464, 0.0942475], [0.344272, 0.423119, -0.303866], \
+      [0.0430714, 0.216233, -0.308475], [0.386085, 0.127333, 0.0503609], \
+      [0.334723, 0.071415, 0.403906]]
+    kernelvalues = [0.10701063104295713, 0.0030052117308328923, 0.003125410084676201, \
```

Comment:
This was calculated using the original (verified) implementation of the kernel in Mathematica, it is used to check if the implementations have identical output